src/lib.rs

```diff
@@ -1475,6 +1478,122 @@ where
+    client_id: Option<ClientId>,
+    post_logout_redirect_uri: Option<RedirectUrl>,
+    state: Option<CsrfToken>,
+    #[serde(serialize_with = "serialize_with_join_vec")]
+    ui_locales: Vec<LanguageTag>,
+}
+
+impl From<EndSessionUrl> for LogoutRequest {
+    fn from(value: EndSessionUrl) -> Self {
+        LogoutRequest {
+            end_session_endpoint: value,
+            parameters: Default::default(),
+        }
+    }
+}
+
+impl LogoutRequest {
+    ///
+    /// Provides an ID token previously issued by this OpenID Connect Provider as a hint about
+    /// the user's identity.
+    ///
+    pub fn set_id_token_hint<AC, GC, JE, JS, JT>(
+        mut self,
+        id_token_hint: &IdToken<AC, GC, JE, JS, JT>,
+    ) -> Self
+    where
+        AC: AdditionalClaims,
+        GC: GenderClaim,
+        JE: JweContentEncryptionAlgorithm<JT>,
+        JS: JwsSigningAlgorithm<JT>,
+        JT: JsonWebKeyType,
+    {
+        self.parameters.id_token_hint = Some(id_token_hint.to_string());
+        self
+    }
+
+    ///
+    /// Provides the OpenID Connect Provider with a hint about the user's identity.
+    ///
+    /// The nature of this hint is specific to each provider.
+    ///
+    pub fn set_logout_hint(mut self, logout_hint: LogoutHint) -> Self {
+        self.parameters.logout_hint = Some(logout_hint);
+        self
+    }
+
+    ///
+    /// Provides the OpenID Connect Provider with the client identifier.
+    ///
+    /// When both this and `id_token_hint` are set, the provider must verify that
+    /// this client id matches the one used when the ID token was issued.
+    ///
+    pub fn set_client_id(mut self, client_id: ClientId) -> Self {
+        self.parameters.client_id = Some(client_id);
+        self
+    }
+
+    ///
+    /// Provides the OpenID Connect Provider with a URI to redirect to after
+    /// the logout has been performed.
+    ///
+    pub fn set_redirect_uri(mut self, redirect_uri: RedirectUrl) -> Self {
+        self.parameters.post_logout_redirect_uri = Some(redirect_uri);
+        self
+    }
+
+    ///
+    /// Specify an opaque value that the OpenID Connect Provider should pass back
+    /// to your application using the state parameter when redirecting to redirect_uri.
+    ///
+    pub fn set_state(mut self, state: CsrfToken) -> Self {
+        self.parameters.state = Some(state);
+        self
+    }
+
+    ///
+    /// Requests the preferred languages for the user interface presented by the OpenID Connect
+    /// Provider.
+    ///
+    /// Languages should be added in order of preference.
+    ///
+    pub fn add_ui_locale(mut self, ui_locale: LanguageTag) -> Self {
+        self.parameters.ui_locales.push(ui_locale);
+        self
+    }
+
+    ///
+    /// Returns the full logout URL
+    ///
+    pub fn url(self) -> Url {
+        let mut url = self.end_session_endpoint.url().to_owned();
+        url.set_query(serde_urlencoded::to_string(self.parameters).ok().as_deref());
```

Comment:
I missed this yesterday, but I don't think we should swallow the serialization error if there is one. hopefully getting rid of `serde_urlencoded` and following `oauth2`'s approach of using `query_pairs_mut()` directly will allow us to have an infallible function, though